.github/workflows/staging-deployment.yml

```diff
@@ -0,0 +1,179 @@
+              -l $DEPLOYMENT_USER $DEPLOYMENT_HOSTS_PRIMARY \
+              "rm -f $DEPLOYMENT_FOLDER/artemis.war.old" || exit 1
+          
+          # Copy new war file
+          scp -o BatchMode=yes -o StrictHostKeyChecking=accept-new \
+              artifacts/*.war $DEPLOYMENT_USER@$DEPLOYMENT_HOSTS_PRIMARY:$DEPLOYMENT_FOLDER/artemis.war.new || {
+            echo "::error::Failed to copy war file to primary host"
+            exit 1
+          }
+
+          # Stop service, rotate files, and start service
+          ssh -o BatchMode=yes -o StrictHostKeyChecking=accept-new \
+              -l $DEPLOYMENT_USER $DEPLOYMENT_HOSTS_PRIMARY << 'ENDSSH' || exit 1
+            set -e
+            sudo systemctl stop artemis
+            mv $DEPLOYMENT_FOLDER/artemis.war $DEPLOYMENT_FOLDER/artemis.war.old
+            mv $DEPLOYMENT_FOLDER/artemis.war.new $DEPLOYMENT_FOLDER/artemis.war
+            sudo systemctl start artemis
+          ENDSSH
+
+      - name: Verify Primary Node Deployment
+        id: verify_deployment
+        run: |
+          while true; do
+            echo "Performing health check..."
+      
+            RESPONSE=$(curl -s -f $HEALTH_CHECK_URL || echo '{"status":"DOWN"}')
+            STATUS=$(echo $RESPONSE | jq -r '.status')
+      
+            if [ "$STATUS" = "UP" ]; then
+              echo "Health check passed! Application is UP"
+              exit 0
+            else
+              echo "Health check failed. Status: $STATUS"
+              echo "Waiting 10 seconds before next attempt..."
+              sleep 10
+            fi
+          done
+
+      - name: Phase 2 - Deploy to Secondary Nodes
+        run: |
+          for node in $DEPLOYMENT_HOSTS_SECONDARY; do
+            echo "[INFO] Deploy on ${node} ..."
+            
+            # Remove old backup
+            ssh -o BatchMode=yes -o StrictHostKeyChecking=accept-new \
+                -l $DEPLOYMENT_USER $node \
+                "rm -f $DEPLOYMENT_FOLDER/artemis.war.old" || continue
+            
+            # Copy new war file
+            scp -o BatchMode=yes -o StrictHostKeyChecking=accept-new \
+                artifacts/*.war $DEPLOYMENT_USER@$node:$DEPLOYMENT_FOLDER/artemis.war.new || {
+              echo "::warning::Failed to copy war file to secondary host ${node}"
+              continue
+            }
+            
+            # Deploy
+            ssh -o BatchMode=yes -o StrictHostKeyChecking=accept-new \
+                -l $DEPLOYMENT_USER $node << 'ENDSSH' || {
+              echo "::warning::Failed to deploy to secondary host ${node}"
+              continue
+            }
+              set -e
+              sudo systemctl stop artemis
+              mv $DEPLOYMENT_FOLDER/artemis.war $DEPLOYMENT_FOLDER/artemis.war.old
+              mv $DEPLOYMENT_FOLDER/artemis.war.new $DEPLOYMENT_FOLDER/artemis.war
+              sudo systemctl start artemis
+            ENDSSH
+          done
```

Comment:
I'd be great if in either this or a future PR you could use/integrate this into the [`artemis-server-cli` script](https://github.com/ls1intum/Artemis/blob/develop/artemis-server-cli). That is meant as a script to be used to do simple admin operations for an Artemis instance. The testserver deployment also uses that.
It would also reduce the current code duplication present in the scripts